master/internal/api_generic_tasks.go

```diff
@@ -0,0 +1,317 @@
+				errors.Wrapf(err,
+					"unable to get external user token").Error())
+		}
+	} else {
+		token, err = user.StartSession(ctx, userModel)
+		if err != nil {
+			return nil, nil, nil, status.Errorf(codes.Internal,
+				errors.Wrapf(err,
+					"unable to create user session inside task").Error())
+		}
+	}
+	taskSpec.UserSessionToken = token
+
+	genericTaskSpec.Base = taskSpec
+	genericTaskSpec.GenericTaskConfig = taskConfig
+
+	genericTaskSpec.Base.ExtraEnvVars = map[string]string{
+		"DET_TASK_TYPE": string(model.TaskTypeGeneric),
+	}
+
+	return genericTaskSpec, launchWarnings, contextDirectoryBytes, nil
+}
+
+func (a *apiServer) canCreateGenericTask(ctx context.Context, projectID int) error {
+	userModel, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return err
+	}
+
+	errProjectNotFound := api.NotFoundErrs("project", fmt.Sprint(projectID), true)
+	p := &projectv1.Project{}
+	if err := a.m.db.QueryProto("get_project", p, projectID); errors.Is(err, db.ErrNotFound) {
+		return errProjectNotFound
+	} else if err != nil {
+		return err
+	}
+	if err := project.AuthZProvider.Get().CanGetProject(ctx, *userModel, p); err != nil {
+		return authz.SubIfUnauthorized(err, errProjectNotFound)
+	}
+
+	if err := command.AuthZProvider.Get().CanCreateGenericTask(
+		ctx, *userModel, model.AccessScopeID(p.WorkspaceId)); err != nil {
+		return status.Errorf(codes.PermissionDenied, err.Error())
+	}
+
+	return nil
+}
+
+func (a *apiServer) CreateGenericTask(
+	ctx context.Context, req *apiv1.CreateGenericTaskRequest,
+) (*apiv1.CreateGenericTaskResponse, error) {
+	var projectID int
+	if req.ProjectId != nil {
+		projectID = int(*req.ProjectId)
+	} else {
+		projectID = model.DefaultProjectID
+	}
+
+	if err := a.canCreateGenericTask(ctx, projectID); err != nil {
+		return nil, err
+	}
+
+	genericTaskSpec, warnings, contextDirectoryBytes, err := a.getGenericTaskLaunchParameters(
+		ctx, req.ContextDirectory, req.Config, projectID,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := check.Validate(genericTaskSpec.GenericTaskConfig); err != nil {
+		return nil, status.Errorf(
+			codes.InvalidArgument,
+			"invalid generic task config: %s",
+			err.Error(),
+		)
+	}
+
+	// Persist the task.
+	taskID := model.NewTaskID()
+	jobID := model.NewJobID()
+	startTime := time.Now()
+	err = db.Bun().RunInTx(ctx, nil, func(ctx context.Context, tx bun.Tx) error {
+		if err := db.AddJobTx(ctx, tx, &model.Job{
+			JobID:   jobID,
+			JobType: model.JobTypeGeneric,
+			OwnerID: &genericTaskSpec.Base.Owner.ID,
+		}); err != nil {
+			return fmt.Errorf("persisting job %v: %w", taskID, err)
+		}
+
+		if err := db.AddTaskTx(ctx, tx, &model.Task{
+			TaskID:     taskID,
+			TaskType:   model.TaskTypeGeneric,
+			StartTime:  startTime,
+			JobID:      &jobID,
+			LogVersion: model.CurrentTaskLogVersion,
+		}); err != nil {
+			return fmt.Errorf("persisting task %v: %w", taskID, err)
+		}
+
+		// TODO persist config elemnts
```

Comment:
plan to do this in 

```
Backend API to query generic tasks
 (including persisting and retrieving Config)
 ```

Reply:
PR for it here: https://github.com/determined-ai/determined/pull/8435